diff --git a/modules/enhanced_editor_embed/src/Controller/PreviewEmbedController.php b/modules/enhanced_editor_embed/src/Controller/PreviewEmbedController.php
index 74b859209..1a104dfb2 100644
--- a/modules/enhanced_editor_embed/src/Controller/PreviewEmbedController.php
+++ b/modules/enhanced_editor_embed/src/Controller/PreviewEmbedController.php
@@ -56,7 +56,7 @@ public function preview(Request $request): Response {
       '#text' => $text,
       '#format' => 'enhanced_editor_embed',
     ];
-    $html = $this->renderer->renderPlain($build);
+    $html = $this->renderer->renderPlain($build)->jsonSerialize();

     // Note that we intentionally do not use:
     // - \Drupal\Core\Cache\CacheableResponse because caching it on the server
