diff --git a/js/plugins/drupalmedia/plugin.es6.js b/js/plugins/drupalmedia/plugin.es6.js
index 6921db5af..4f8c184b3 100644
--- a/js/plugins/drupalmedia/plugin.es6.js
+++ b/js/plugins/drupalmedia/plugin.es6.js
@@ -132,7 +132,7 @@

         editables: {
           caption: {
-            selector: 'figcaption',
+            selector: 'figcaption,smc-figure > *[slot="caption"]',
             allowedContent: 'a[!href]; em strong cite code br',
             pathName: Drupal.t('Caption'),
           },
@@ -323,7 +323,7 @@

           // Find the actual embedded media in the DOM.
           const embeddedMediaContainer = this.data.hasCaption
-            ? this.element.findOne('figure')
+            ? this.element.findOne('figure,smc-figure')
             : this.element;
           let embeddedMedia = embeddedMediaContainer.getFirst(isElementNode);
           // If there is a link, the top-level element is the `a` tag, and the
