diff --git a/core/modules/ckeditor/js/plugins/drupalmedia/plugin.es6.js b/core/modules/ckeditor/js/plugins/drupalmedia/plugin.es6.js
index 7a0c5a4213..4f8c184b33 100644
--- a/core/modules/ckeditor/js/plugins/drupalmedia/plugin.es6.js
+++ b/core/modules/ckeditor/js/plugins/drupalmedia/plugin.es6.js
@@ -132,7 +132,7 @@
 
         editables: {
           caption: {
-            selector: 'figcaption',
+            selector: 'figcaption,smc-figure > *[slot="caption"]',
             allowedContent: 'a[!href]; em strong cite code br',
             pathName: Drupal.t('Caption'),
           },
@@ -222,6 +222,7 @@
               widget._setUpDynamicEditables();
               widget._setUpEditButton();
               editor.fire('unlockSnapshot');
+              editor.fire('saveSnapshot');
             });
           }
 
@@ -322,7 +323,7 @@
 
           // Find the actual embedded media in the DOM.
           const embeddedMediaContainer = this.data.hasCaption
-            ? this.element.findOne('figure')
+            ? this.element.findOne('figure,smc-figure')
             : this.element;
           let embeddedMedia = embeddedMediaContainer.getFirst(isElementNode);
           // If there is a link, the top-level element is the `a` tag, and the
diff --git a/core/modules/ckeditor/js/plugins/drupalmedia/plugin.js b/core/modules/ckeditor/js/plugins/drupalmedia/plugin.js
index 8a8e1d574b..c0a5266ea9 100644
--- a/core/modules/ckeditor/js/plugins/drupalmedia/plugin.js
+++ b/core/modules/ckeditor/js/plugins/drupalmedia/plugin.js
@@ -87,7 +87,7 @@
         pathName: Drupal.t('Embedded media'),
         editables: {
           caption: {
-            selector: 'figcaption',
+            selector: 'figcaption,smc-figure > *[slot="caption"]',
             allowedContent: 'a[!href]; em strong cite code br',
             pathName: Drupal.t('Caption')
           }
@@ -118,7 +118,12 @@
               }
             });
           }
-          var hostEntityLangcode = document.getElementById(editor.name).getAttribute('data-media-embed-host-entity-langcode');
+          var hostEntityLangcode = '';
+          if (document.getElementById(editor.name)){
+            hostEntityLangcode = document.getElementById(editor.name).getAttribute('data-media-embed-host-entity-langcode');
+          } else {
+            hostEntityLangcode = 'en';
+          }
           if (hostEntityLangcode) {
             data.hostEntityLangcode = hostEntityLangcode;
           }
@@ -143,6 +148,7 @@
               widget._setUpDynamicEditables();
               widget._setUpEditButton();
               editor.fire('unlockSnapshot');
+              editor.fire('saveSnapshot');
             });
           }
           if (this.oldData) {
@@ -190,7 +196,7 @@
           var isElementNode = function isElementNode(n) {
             return n.type === CKEDITOR.NODE_ELEMENT;
           };
-          var embeddedMediaContainer = this.data.hasCaption ? this.element.findOne('figure') : this.element;
+          var embeddedMediaContainer = this.data.hasCaption ? this.element.findOne('figure,smc-figure') : this.element;
           var embeddedMedia = embeddedMediaContainer.getFirst(isElementNode);
           if (this.data.link) {
             embeddedMedia = embeddedMedia.getFirst(isElementNode);
